File: little-bee/controller/router/driver_type.go
package router

import (
	"controller/config"

	"github.com/gin-gonic/gin"
)

const URL_ROUTER_DRIVER_TYPE = "/config/driver-type"
const URL_ROUTER_DRIVER_TYPE_BY_ID = "/config/driver-type/:driver-type-id"
const URL_ROUTER_DRIVER_TYPE_QUERY = "/config/driver-type/query"
const URL_ROUTER_DRIVER_TYPE_ALL = "/config/driver-type/all"

func DriverTypeRouter(r *gin.RouterGroup) *gin.RouterGroup {
	r.POST(URL_ROUTER_DRIVER_TYPE, config.CreateDriverType)
	r.GET(URL_ROUTER_DRIVER_TYPE_BY_ID, config.RetrieveDriverType)
	r.PUT(URL_ROUTER_DRIVER_TYPE, config.UpdateDriverType)
	r.DELETE(URL_ROUTER_DRIVER_TYPE_BY_ID, config.DeleteDriverType)
	r.PUT(URL_ROUTER_DRIVER_TYPE_BY_ID, config.DeleteDriverTypeWithFlag)
	r.POST(URL_ROUTER_DRIVER_TYPE_QUERY, config.QueryDriverType)
	r.GET(URL_ROUTER_DRIVER_TYPE_ALL, config.GetAllDriverTypes)
	return r
}

File: frontend/console/src/api/driver-type.js
import Axios from "@/http";

export function createDriverType(params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/driver-type', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function retrieveDriverType(driverTypeId) {
    return new Promise((resolve, reject) => {
        Axios.get(`/config/driver-type/${ driverTypeId }`).then(res => {
            resolve(res)
        })
    })
}

export function updateDriverType(params) {
    return new Promise((resolve, reject) => {
        Axios.put('/config/driver-type', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function deleteDriverType(driverTypeId) {
    return new Promise((resolve, reject) => {
        Axios.delete(`/config/driver-type/${ driverTypeId }`).then(res => {
            resolve(res)
        })
    })
}

export function deleteDriverTypeWithFlag(driverTypeId) {
    return new Promise((resolve, reject) => {
        Axios.put(`/config/driver-type/${ driverTypeId }`).then(res => {
            resolve(res)
        })
    })
}

export function getAllDriverTypes() {
    return new Promise((resolve, reject) => {
        Axios.get('/config/driver-type/all').then(res => {
            resolve(res)
        })
    })
}

export function queryDriverType (params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/driver-type/query', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}